### PR TITLE
handle adding new score

### PR DESCRIPTION
**Pull request by mikitachab**
- added forms and handlers for creating new score
- not finished, we need to implement server side validation rules and error handling
- not tested all new handlers because of complicity and they are should be changed in near future

this PR can be considered as solution for #13

### CODE REVIEW

File: server/views.go [hunk 1]
```diff
@@ -53,3 +54,55 @@ func handleAddPlayer(ctx *HandlerCtx) http.HandlerFunc {
...
+		"Civilian Structures",
+		"Scientific Structures",
+		"Commercial Structures",
+		"Guilds",
+	}
+	return func(w http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case "GET":
+			players := r.URL.Query()["player"]
+			ctx.Template.Render(w, struct {
```
Comment (antonskwr) [inline server/views.go]: This is a little unreadable, consider using something like:
```
var dataWrapper struct {
	Players []string
	Categories []string
}

dataWrapper.Players = players
dataWrapper.Categories = categories
```

File: server/views.go [hunk 2]
```diff
@@ -53,3 +54,55 @@ func handleAddPlayer(ctx *HandlerCtx) http.HandlerFunc {
...
+				Categories []string
+			}{players, categories})
+		case "POST":
+			r.ParseForm()
+			scores := GetScoresFromForm(r.Form)
+			for k, v := range scores {
+				fmt.Printf("%s %v", k, v)
+			}
+			ctx.DB.SaveScores(scores)
+			http.Redirect(w, r, "/", 302)
```
Comment (antonskwr) [inline server/views.go]: Consider using `http.StatusFound` instead of `302`

File: template/add_score_select_players.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{{ define "view" }}
```
Comment (antonskwr) [inline template/add_score_select_players.html]: Trim white spaces for consistency

File: template/add_score_select_players.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+                <input class="form-check-input" type="checkbox" id="{{.Name}}" name="{{.Name}}">
+                <label class="form-check-label" for="{{.Name}}">{{.Name}}</label>
+            </div>
+        {{end}}
+        </div>
+        <button type="submit" class="btn btn-primary my-2">Submit</button>
+    </form>
+</div>
+
+{{ end }}
```
Comment (antonskwr) [inline template/add_score_select_players.html]: Trim white spaces for consistency

File: server/views.go [hunk 5]
```diff
@@ -53,3 +54,55 @@ func handleAddPlayer(ctx *HandlerCtx) http.HandlerFunc {
...
+			addScoreURL, err := url.Parse("/score/add/")
+			if err != nil {
+				panic(err)
+			}
+			q := addScoreURL.Query()
+			for name := range r.Form {
+				q.Add("player", name)
+			}
+			addScoreURL.RawQuery = q.Encode()
+			http.Redirect(w, r, addScoreURL.String(), 302)
```
Comment (antonskwr) [inline server/views.go]: Consider using `http.StatusFound` instead of `302`

**Review by antonskwr — CHANGES_REQUESTED** [review]

File: server/views.go (hunk 1 above)
Comment (mikitachab) [reply server/views.go]: done

File: server/views.go (hunk 2 above)
Comment (mikitachab) [reply server/views.go]: done

File: template/add_score_select_players.html (hunk 3 above)
Comment (mikitachab) [reply template/add_score_select_players.html]: done

File: template/add_score_select_players.html (hunk 4 above)
Comment (mikitachab) [reply template/add_score_select_players.html]: done

File: server/views.go (hunk 5 above)
Comment (mikitachab) [reply server/views.go]: done

**Review by antonskwr — APPROVED** [review]

**Pull request merged**